web/src/beta/features/PluginPlayground/Plugins/hook.ts

```diff
@@ -0,0 +1,188 @@
+import JSZip from "jszip";
+import { useCallback, useMemo, useState } from "react";
+import useFileInput from "use-file-input";
+
+import { PluginType, REEARTH_YML_FILE } from "./constants";
+import { validateFileTitle } from "./utils";
+
+export default () => {
+  const [plugins, setPlugins] = useState<PluginType[]>([
+    {
+      id: generateUniqueId(),
+      title: "My Plugin",
+      files: [REEARTH_YML_FILE]
+    }
+  ]);
+
+  const [selectedPluginId, setSelectedPluginId] = useState(plugins[0].id);
+  const [selectedFileId, setSelectedFileId] = useState<string>(
+    REEARTH_YML_FILE.id
+  );
+
+  const selectedPlugin = useMemo(
+    () =>
+      plugins.find((plugin) => plugin.id === selectedPluginId) ?? plugins[0],
+    [plugins, selectedPluginId]
+  );
+
+  const selectedFile = useMemo(
+    () =>
+      selectedPlugin.files.find((file) => file.id === selectedFileId) ??
+      selectedPlugin.files[0],
+    [selectedPlugin, selectedFileId]
+  );
+
+  const selectPlugin = useCallback((pluginId: string) => {
+    setSelectedPluginId(pluginId);
+  }, []);
+
+  const selectFile = useCallback((fileId: string) => {
+    setSelectedFileId(fileId);
+  }, []);
+
+  const addFile = useCallback(
+    (title: string) => {
+      const result = validateFileTitle(
+        title,
+        selectedPlugin.files.map((f) => f.title)
+      );
+      if (!result.success) {
+        return;
+      }
+      const newFile = {
+        id: generateUniqueId(),
+        title,
+        sourceCode: ""
+      };
+
+      setPlugins((plugins) =>
+        plugins.map((plugin) =>
+          plugin.id === selectedPlugin.id
+            ? { ...plugin, files: [...plugin.files, newFile] }
+            : plugin
+        )
+      );
+
+      setSelectedFileId(newFile.id);
+    },
+    [selectedPlugin]
+  );
+
+  const updateFileTitle = useCallback(
+    (newTitle: string, id: string) => {
+      const result = validateFileTitle(
+        newTitle,
+        selectedPlugin.files.map((f) => f.title)
+      );
+
+      if (!result.success) {
+        return;
+      }
```

Comment:
We can make sure of the `useNotification` hook to create notifications when a file title is not value or any process is unsuccessful so that the user is informed of the error